lizmap/modules/lizmap/lib/Request/WMSRequest.php

```diff
@@ -265,13 +265,21 @@ protected function checkMaximumWidthHeight()
      *
      * @see https://en.wikipedia.org/wiki/Web_Map_Service#Requests.
      */
-    protected function process_getlegendgraphic()
+    protected function process_getlegendgraphics()
     {
-        return $this->process_getlegendgraphics();
+        return $this->process_getlegendgraphic();
     }
 
-    protected function process_getlegendgraphics()
+    protected function process_getlegendgraphic()
     {
+        // The right request name is GetLegendGraphic not GetLegendGraphics
+        $wmsRequest = strtolower($this->param('request'));
+        if ($wmsRequest == 'getlegendgraphic') {
+            $this->params = array_merge(
+                $this->params,
+                Proxy::normalizeParams(array('request' => 'getlegendgraphic'))
```

Comment:
Params were normalized by OGCRequest constructor

https://github.com/3liz/lizmap-web-client/blob/74dee7ddfd979f40d331305b3caebd026603f97e/lizmap/modules/lizmap/lib/Request/OGCRequest.php#L77

why not
```php
$this->params['request'] = 'getlegendgraphic';
```

Reply:
you are right @nworr